### PR TITLE
InfluxDB: variables migration to backend

### PR DESCRIPTION
**Pull request by joey-grafana**
**What this PR does / why we need it**:
Part of https://github.com/grafana/grafana/issues/43076

This PR contains the changes needed to applyTemplateVariables() to InfluxQL queries (which are run through the backend).

**Test scenarios**:

1. Run make devenv sources=influxdb,influxdb1,grafana grafana_version=8.2.4 (or other version before changes)
2. Open up http://localhost:3000/explore && http://127.0.0.1:3001/explore (to test older 8.2.4 version)
3. Select gdev-influxdb-influxql
    - applyTemplateVariables() should apply all template variables in explore queries that are run through the backend via `api/ds/query`
4. Select gdev-influxdb1-influxql
    - Should behave like `gdev-influxdb-influxql`

PS: make sure 
[feature_toggles]
enable = influxdbBackendMigration

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 1]
```diff
@@ -153,6 +153,9 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
   }
 
   query(request: DataQueryRequest<InfluxQuery>): Observable<DataQueryResponse> {
+    // backward compatibility
+    request.scopedVars.interval = request.scopedVars.__interval;
```
[the remark this answers is not in the document]
Comment (joey-grafana) [reply public/app/plugins/datasource/influxdb/datasource.ts]: Does not need to be copied over from classicQuery (I thought we would need it but I can't see any cases where that's true). Removing.

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 2]
```diff
@@ -381,67 +384,67 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
...
-              params: groupBy.params?.map((param) => {
-                return this.templateSrv.replace(param.toString(), undefined, 'regex');
-              }),
-            };
-          });
-        }
+  applyVariables(query: InfluxQuery, scopedVars: ScopedVars, rest: ScopedVars) {
+    if (query.groupBy) {
+      query.groupBy = query.groupBy.map((groupBy) => {
+        return {
```
Comment (yesoreyeram) [inline public/app/plugins/datasource/influxdb/datasource.ts:407]: I have seen problems before when mutating the query direcly like this. But can't remember what. Can we use the original method like `const expandedQuery = { ...query }` and then return the copy?

@gabor

**Review by yesoreyeram — APPROVED** [review]
I am ok to approve. But prefer to use @gabor's expertise as well.

File: public/app/plugins/datasource/influxdb/datasource.ts (hunk 2 above)
Comment (gabor) [reply public/app/plugins/datasource/influxdb/datasource.ts:407]: i would also recommend not mutating the query (i don't have any specific problem in mind that could be caused by this, but in general we should not mutate what we do not create).

we have https://immerjs.github.io/immer/ in our `package.json`, i recommend using it here if the `{...query}` approach gets too long.

Comment (joey-grafana) [reply public/app/plugins/datasource/influxdb/datasource.ts:408]: Thanks for the tip.. pushed changes.

**Pull request merged**